packages/core/src/shared/context.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AsyncLocalStorage } from './telemetry/spans'
+
+/**
+ * Returns the latest execution context.
+ *
+ * Ensure your functions are wrapping their code with {@link withExecutionContext} to actually
+ * add the execution context to the call stack. Otherwise, this will be missing data.
+ */
+export function getCurrentExecutionContext(): ExecutionContextStore | undefined {
+    return executionContext.getStore()
+}
+
+/**
+ * Runs the given callback with the additional provided execution context.
+ * I.e we can recursively know the method that executed the current method. This gives an idea of
+ * who called who during execution.
+ *
+ * Finally, use {@link getCurrentExecutionContext()} to get the context when required.
+ *
+ * See the tests for examples of how to use this.
+ *
+ * @param store.source An identifier that represents the callback. You'll probably want to use the function name.
+ * @param store.source If the source is a method, you'll want to include the class name for better context.
+ * @param callback The function to run within this execution context. You'll probably want your entire function's code
+ *                 to be wrapped in this.
+ */
+export const withExecutionContext = (context: { source: string; class?: string }, callback: () => any) =>
+    executionContext.run(createStore(context), callback)
+
+type ExecutionContextStore = {
+    /**
+     * The stringified representation of the call stack.
+     *
+     * Example: "ClassA#methodA,methodB:ClassB#methodY,methodZ"
```

Comment:
How is a method without a class represented?

Reply:
I've updated the description again. But it will just not have a `ClassX#` in front of it.

Eg: `ClassA#methodA:myFunctionWithNoClass`